x-pack/plugins/maps/public/classes/styles/vector/components/legend/vector_style_legend.tsx

```diff
@@ -33,12 +36,47 @@ export function VectorStyleLegend({
   const legendRows = [];
 
   for (let i = 0; i < styles.length; i++) {
-    const row = styles[i].renderLegendDetailRow({
-      isLinesOnly,
-      isPointsOnly,
-      symbolId,
-      svg,
-    });
+    const styleMetaDataRequest = styles[i].isDynamic()
+      ? (styles[i] as DynamicStyleProperty<DataRequest>).getStyleMetaDataRequest()
+      : undefined;
+
+    const error = styleMetaDataRequest?.getError();
+    const fieldName = (styles[i] as DynamicStyleProperty<DataRequest>)?.getFieldName();
```

Comment:
Instead of fieldName, could you use `field.getLabel`. Also, `field.getLabel` is async so it may make sense to move error rendering into a separate component.

```
const field = this.props.style.getField();
    if (!field) {
      return;
    }
    const label = await field.getLabel();
```